src/reports/SelfServiceReports.js

```diff
@@ -11,81 +10,191 @@ import { makeStyles } from "@material-ui/core";
 import Grid from "@material-ui/core/Grid";
 import MetabaseSVG from "./Metabase_icon.svg";
 import Button from "@material-ui/core/Button";
+import Switch from "@material-ui/core/Switch";
+import CircularProgress from "@material-ui/core/CircularProgress";
 
 const useStyles = makeStyles({
   root: {
-    maxWidth: 500,
+    maxWidth: 600,
     backgroundColor: "#FFF"
   },
-  details: {
-    display: "flex"
+  topRightButtons: {
+    display: "flex",
+    justifyContent: "flex-end",
+    gap: "10px",
+    marginTop: "10px"
+  },
+  actionAreaContainer: {
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center"
   },
-  cover: {
-    width: 151
+  buttonSpacing: {
+    marginLeft: "10px"
   },
-  content: {
-    flex: "1 0 auto"
+  message: {
+    color: "green",
+    marginTop: "10px"
+  },
+  errorMessage: {
+    color: "red",
+    marginTop: "10px"
   }
 });
 
 const SelfServiceReports = () => {
   const classes = useStyles();
+  const [setupEnabled, setSetupEnabled] = useState(false);
+  const [loadingSetup, setLoadingSetup] = useState(false);
+  const [loadingRefresh, setLoadingRefresh] = useState(false);
+  const [syncStatus, setSyncStatus] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timer = setTimeout(() => {
+        setErrorMessage("");
```

Comment:
No need to hide error message after 10 seconds. Use a toast if you want to do this.

Reply:
Alright will try to use toast !